src/content/peerconnection/endtoend-encryption/js/main.js

```diff
@@ -110,6 +112,23 @@ function dump(chunk, direction, max = 16) {
   console.log(performance.now().toFixed(2), direction, bytes.trim(), chunk.data.byteLength, (data[0] & 0x1) === 0);
 }
 
+// If using crypto offset (controlled by a checkbox):
+// Do not encrypt the first couple of bytes of the payload. This allows
+// a middle to determine video keyframes or the opus mode being used.
+// For VP8 this is the content described in
+//   https://tools.ietf.org/html/rfc6386#section-9.1
+// which is 10 bytes for key frames and 3 bytes for delta frames.
+// For opus (where chunk.type is not set) this is the TOC byte from
+//   https://tools.ietf.org/html/rfc6716#section-3.1
+//
+// It makes the (encrypted) video and audio much more fun to watch and listen to
+// as the decoder does not immediately throw a fatal error.
+const frameTypeToCryptoOffset = {
+  key: 10,
+  delta: 3,
+  undefined: 1,
```

Comment:
I would very much prefer being able to determine audio/video in a way that differs from "does the frame have a type"

Reply:
You can determine it when installing the function - at that time you know the track type.